modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java

```diff
@@ -384,6 +384,12 @@
      */
     private boolean encryptionEnabled;
 
+    /** */
+    private PageCompression pageCompression;
```

Comment:
Subject to discuss: given that hole-punching will not be the only compression approach, should we:
1) Introduce some sort of compression strategy?
2) Add an ability to have a pluggable compression algorithm?
3) For hole-punching in-memory caches are not supported. For some other compression approaches we may support in-memory caches, but data from different caches may be placed to the same page. Should we move this to the DataStorageConfiguration?